File: Data Structures & Algorithms/is-palindrome/submission-1.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string temp="";
        for(int i=0;i<s.length();i++){
            if(('a'<=s[i] && 'z'>=s[i])|| (s[i]>='0' && s[i]<='9'))temp+=s[i];
            else if('A'<=s[i] && 'Z'>=s[i])temp+=(s[i]+32);
            else continue;
        }

        int i=0,j=temp.length()-1;
        while(i<j)if(temp[i]!=temp[j])return false;
        else {
            i++;j--;
        }
        return true;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-1.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen =set()        
        while n!=1 and n not in seen:
            seen.add(n)
            temp=0
            while n>0:
                t=n%10
                temp+=(t*t)
                n//=10
            n=temp

        return True if n==1 else False

        